option.go

```diff
@@ -255,6 +256,18 @@ func OptionHide() func(*BaseRoute) {
 
 func OptionSecurity(securityRequirements ...openapi3.SecurityRequirement) func(*BaseRoute) {
 	return func(r *BaseRoute) {
+
+		// Validate the security scheme exists in components
+		for _, req := range securityRequirements {
+			for schemeName := range req {
+				if r.mainRouter.OpenApiSpec.Components != nil {
```

Comment:
Is it really possible for this to be nil? And in the case that it is would really want the `SecurityRequirement`s to be valid

Reply:
Right, so maybe something like this would look a bit neater

```golang
func OptionSecurity(securityRequirements ...openapi3.SecurityRequirement) func(*BaseRoute) {
	return func(r *BaseRoute) {
               if r.mainRouter.OpenApiSpec.Components == nil {
                   panic("zero security schemas have been registered with the server")
               }

		// Validate the security scheme exists in components
		for _, req := range securityRequirements {
			for schemeName := range req {
				if _, exists := r.mainRouter.OpenApiSpec.Components.SecuritySchemes[schemeName]; !exists {
					panic(fmt.Sprintf("security scheme '%s' not defined in components", schemeName))
				}
			}
		}
```

and catch the case we're talking about.

Also I hope you're enjoying go 😄